Azure Compute Fleet GA version 2024-11-01 Minor update to Documentation only.

rahuls-microsoft opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Azure Compute Fleet GA version 2024-11-01

rahuls-microsoft committed:
Updating last commitId.

rahuls-microsoft committed:
Updating last commitId.

rahuls-microsoft committed:
Generating SDK from last checked in swagger.

rahuls-microsoft committed:
Generating SDK from last checked in swagger.

rahuls-microsoft committed:
Merge branch 'fleet-sdk-24-11-01'

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Updating CommitID from - https://github.com/Azure/azure-rest-api-specs/commit/0b5555556f3288ee49d3a154cb0662d75b14a4ba

rahuls-microsoft committed:
Updating CommitID from - https://github.com/Azure/azure-rest-api-specs/commit/0b5555556f3288ee49d3a154cb0662d75b14a4ba

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Update the Version.

rahuls-microsoft committed:
Update the Version.

jsquire commented on eng/Packages.Data.props:
```diff
@@ -354,7 +354,7 @@
     <PackageReference Update="Portable.BouncyCastle" Version="1.9.0" />
     <PackageReference Update="PublicApiGenerator" Version="10.0.1" />
-    <PackageReference Update="System.ClientModel" Version="1.2.0" />
+    <PackageReference Update="System.ClientModel" Version="1.2.1" />
```
This change already exists in main and should not appear in this PR.

rahuls-microsoft replied:
Ok, I was getting build errors. I will pull the changes from main.

rahuls-microsoft replied:
I have merged from the source branch, so it should resolve as no change, right?

jsquire replied:
Rebase on main, don't merge.   You want to put your changes on top of the latest.   There should be no files in this PR that you're not changing for your work.

rahuls-microsoft replied:
rebased to https://github.com/Azure/azure-sdk-for-net.git

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.ComputeFleet](https://apiview.dev/Assemblies/Review/1ef08d70c28d430d83e203b1473e53bd?revisionId=e63cf77a901944d9b49c43d4c66622ab)

jsquire reviewed (dismissed):
@rahuls-microsoft :

Your account lacks the required public Azure GitHub organization required of an internal contributor.  Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.

- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

rahuls-microsoft committed:
Spures cspell for word Rdma.

rahuls-microsoft committed:
Spures cspell for word Rdma.

rahuls-microsoft committed:
Suppress - //cspell:ignore Rdma

rahuls-microsoft committed:
Suppress - //cspell:ignore Rdma

rahuls-microsoft committed:
Adding cpell.json to override Rdma spelling.

rahuls-microsoft committed:
Adding cpell.json to override Rdma spelling.

rahuls-microsoft committed:
Adding rahuls-microsoft for fleet owners.

rahuls-microsoft committed:
Adding rahuls-microsoft for fleet owners.

rahuls-microsoft committed:
Merge branch 'main' of https://github.com/rahuls-microsoft/azure-sdk-for-net

rahuls-microsoft commented:
> @rahuls-microsoft :
> 
> Your account lacks the required public Azure GitHub organization required of an internal contributor. Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
> 
> * [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
> * [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_
> 
> You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.
> 
> Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

Added myself to Codeowners.

rahuls-microsoft committed:
Remove local cspell.json and add rdma to global cspell.json.

rahuls-microsoft committed:
Remove local cspell.json and add rdma to global cspell.json.

rahuls-microsoft commented:
> > @rahuls-microsoft :
> > Your account lacks the required public Azure GitHub organization required of an internal contributor. Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
> > 
> > * [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
> > * [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_
> > 
> > You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.
> > Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.
> 
> Added myself to Codeowners. But gives error on adding it. How do add myself to CODEOWNERS - directly checkin to - https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS ?

https://github.com/Azure/azure-sdk-for-net/pull/46554 - Created a pull request to add myself to codeowners for Fleet.

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
restore CodeOwners

rahuls-microsoft committed:
restore CodeOwners

rahuls-microsoft committed:
Merge branch 'main' of https://github.com/rahuls-microsoft/azure-sdk-for-net

rahuls-microsoft commented:
> @rahuls-microsoft :
> 
> Your account lacks the required public Azure GitHub organization required of an internal contributor. Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
> 
> * [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
> * [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_
> 
> You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.
> 
> Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

Added myself to - https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS

rahuls-microsoft commented:
@jsquire - Updated the PR after rebase.
Please review.

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

jsquire commented:
> @jsquire - Updated the PR after rebase. Please review.

Common repository files look good.   You'll need Arthur and team for the review of your library code.   Your group memberships are still incorrect, and this will remain blocked until we fix those.

danieljurek commented on .vscode/cspell.json:
```diff
@@ -174,7 +174,8 @@
     "westus",
     "Wmem",
-    "xunit"
+    "xunit",
+    "Rdma"
```
Please insert this into the list in alphabetical order

rahuls-microsoft replied:
ok

rahuls-microsoft committed:
ng in alphabatical order.

rahuls-microsoft commented:
> > @jsquire - Updated the PR after rebase. Please review.
> 
> Common repository files look good. You'll need Arthur and team for the review of your library code. Your group memberships are still incorrect, and this will remain blocked until we fix those.

I added myself in the main, should I add myself in this PR? Sorry not familiar with the process. Thanks for your help.

rahuls-microsoft committed:
Adding myself to codeOwners for fleet.

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft commented:
The PS scripts gives errors-
PS C:\src\Fleet-NetSDK\sdk\computefleet\Azure.ResourceManager.ComputeFleet> ..\..\..\eng\scripts\Export-API.ps1
At C:\src\Fleet-NetSDK\eng\scripts\Export-API.ps1:19 char:57
+ $diagnosticArguments = ($debugLogging -and $logsFolder) ? "/binarylog ...
+                                                         ~
Unexpected token '?' in expression or statement.
    + CategoryInfo          : ParserError: (:) [], ParseException
    + FullyQualifiedErrorId : UnexpectedToken

PS C:\src\Fleet-NetSDK\sdk\computefleet\Azure.ResourceManager.ComputeFleet> ..\..\..\eng\scripts\CodeChecks.ps1 -ServiceDirectory computefleet
At C:\src\Fleet-NetSDK\eng\scripts\CodeChecks.ps1:97 char:65
+ ...     $diagnosticArguments = ($debugLogging -and $logsFolder) ? "/binar ...
+                                                                 ~
Unexpected token '?' in expression or statement.
    + CategoryInfo          : ParserError: (:) [], ParseException
    + FullyQualifiedErrorId : UnexpectedToken
    
    Any help on this?

rahuls-microsoft commented:
> > > @jsquire - Updated the PR after rebase. Please review.
> > 
> > 
> > Common repository files look good. You'll need Arthur and team for the review of your library code. Your group memberships are still incorrect, and this will remain blocked until we fix those.
> 
> I added myself in the main, should I add myself in this PR? Sorry not familiar with the process. Thanks for your help.

https://github.com/Azure/azure-sdk-for-net/pull/46575 - Created a PR to add myself to CODEOWNERS

rahuls-microsoft committed:
Adding to CodeOwners for fleet.

danieljurek commented on .vscode/cspell.json:
```diff
@@ -141,6 +141,7 @@
     "referer",
     "renormalize",
     "retriable",
+	"Rdma",
```
```suggestion
    "Rdma",
```

rahuls-microsoft commented:
@ArthurMa1978 - Please suggest what needs to be updated.
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/computefleet/Azure.ResourceManager.ComputeFleet/assets.json#L5C11-L5C73 - What new GUID goes in there?

rahuls-microsoft committed:
Update .vscode/cspell.json

Co-authored-by: Daniel Jurek <djurek@microsoft.com>

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
update changelog.

rahuls-microsoft commented:
> @rahuls-microsoft :
> 
> Your account lacks the required public Azure GitHub organization required of an internal contributor. Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
> 
> * [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
> * [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_
> 
> You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.
> 
> Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

Made the Azure profile public and submitted Pr to Add to CodeOwners - https://github.com/Azure/azure-sdk-for-net/pull/46575
![image](https://github.com/user-attachments/assets/b0dcdc6f-eb6f-4003-b84c-e1e21e0c329b)

rahuls-microsoft committed:
adding  @Azure/compute-fleet Group for fleet team.

rahuls-microsoft committed:
Remove compute-fleet group from codeowners.

rahuls-microsoft committed:
Merge branch 'main' into main

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

live1206 commented on sdk/computefleet/Azure.ResourceManager.ComputeFleet/api/Azure.ResourceManager.ComputeFleet.netstandard2.0.cs:
```diff
@@ -1,5 +1,6 @@
 namespace Azure.ResourceManager.ComputeFleet
 {
+	//cspell:ignore Rdma
```
The API file is auto-generated, should not edit it.

rahuls-microsoft replied:
ok, will remove it.

rahuls-microsoft replied:
ok

live1206 commented on .vscode/cspell.json:
```diff
@@ -141,6 +141,7 @@
     "referer",
     "renormalize",
     "retriable",
+    "Rdma",
```
probably add this only for your service, like L182-L186

live1206 replied:
This is not updated though

rahuls-microsoft replied:
Will move to service specific section, as suggested.- 
{
      "filename": "**/sdk/computefleet/**/*.cs",
      "words": [
        "Rdma"
      ]
    },

rahuls-microsoft replied:
Updated as suggested.

live1206 commented:
Fix the CI failure, guide can be found in https://github.com/Azure/azure-sdk-for-net/pull/46541/checks?check_run_id=31491180031

live1206 commented on sdk/computefleet/Azure.ResourceManager.ComputeFleet/CHANGELOG.md:
```diff
@@ -1,14 +1,32 @@
+    - Better error-handling.
+    - Support uniform telemetry across all languages.
+
+## 1.0.0-beta.2 (2023-11)
```
remove this entry since you already added a new GA entry

rahuls-microsoft replied:
Ok, will remove - ## 1.0.0-beta.2 (2023-11)

rahuls-microsoft committed:
Remove 1.0.0-beta.2 (2023-11), as suggested.

rahuls-microsoft committed:
Remove - //cspell:ignore Rdma

live1206 commented:
Since this is going GA, we should have a full test coverage for all operations, such as 
https://github.com/Azure/azure-sdk-for-net/blob/fb0b80d4c8fc99e9e964862f2a374d2bc28096f7/sdk/computefleet/Azure.ResourceManager.ComputeFleet/src/Generated/ComputeFleetResource.cs#L373C56-L373C87
Add more tests in scenario test beside https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/computefleet/Azure.ResourceManager.ComputeFleet/tests/Scenario/ComputeFleetCRUDTests.cs

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Merge branch 'main' of https://github.com/rahuls-microsoft/azure-sdk-for-net

rahuls-microsoft committed:
remove - //cspell:ignore Rdma

jsquire commented on .github/CODEOWNERS:
```diff
@@ -1070,9 +1070,9 @@
 
 # PRLabel: %Compute - Fleet
-/sdk/computefleet/Azure.ResourceManager.*/                         @sahilarora92 @dhruvil009 @rahuls-microsoft
+/sdk/computefleet/                                                 @sahilarora92 @dhruvil009 @rahuls-microsoft
```
Please revert this file.   It was properly updated in a separate PR.

rahuls-microsoft replied:
Restored to head - git restore -s HEAD .github\CODEOWNERS

jsquire replied:
Nope.  Still showing changes.   This file should not be included in the change set.

rahuls-microsoft replied:
Ok, what command should be used?

rahuls-microsoft replied:
I used -
git remote add mainSource https://github.com/rahuls-microsoft/azure-sdk-for-net.git
git fetch mainSource
git checkout main
git reset --hard mainSource/main  
git push origin main --force 

Is that fine?
How do I check it, as you checked? So, I can validate it before resolving it.

live1206 replied:
I think the easiest way to revert is
- pull latest main to your local
- then run "git checkout origin/main .github/CODEOWNERS" under the repo root folder

rahuls-microsoft replied:
Also execute - git rebase mainSource/main - This should resolve the issue.
Else please let me know. Thanks

rahuls-microsoft replied:
> I think the easiest way to revert is
> 
> * fetch/pull latest origin/main to your local
> * switch to your working branch
> * then run "git checkout origin/main .github/CODEOWNERS" under the repo root folder

I have created a fork - https://github.com/rahuls-microsoft/azure-sdk-for-net/.
SO, taking too long time and showing lots of conflicts. Is there an easier way or should I just create branch from - https://github.com/Azure/azure-sdk-for-net. That may be an easier option.
Please suggest. Thanks for your help!

live1206 replied:
I reverted it for you

rahuls-microsoft replied:
Thanks

rahuls-microsoft replied:
Sorry! How do I undo it?

live1206 replied:
I redid it.
Merge the latest from remote before you push to remote, instead of force push, it is error-prune with force push.

rahuls-microsoft replied:
@live1206 - Please apply the changes. Thanks for yourr help.

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Remove --pre-release flag from instructions.

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

live1206 committed:
revert CODEOWNERS

rahuls-microsoft committed:
Moving Rdma to computefleet service, as suggested.

live1206 committed:
Merge branch 'main' of https://github.com/rahuls-microsoft/azure-sdk-for-net into rahuls-microsoft/main

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Updates.

rahuls-microsoft committed:
Merge branch 'main' of https://github.com/rahuls-microsoft/azure-sdk-for-net

melina5656 committed:
Change the name of the Property

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Adding more tests.

rahuls-microsoft commented:
> Since this is going GA, we should have a full test coverage for all operations, such as https://github.com/Azure/azure-sdk-for-net/blob/fb0b80d4c8fc99e9e964862f2a374d2bc28096f7/sdk/computefleet/Azure.ResourceManager.ComputeFleet/src/Generated/ComputeFleetResource.cs#L373C56-L373C87 Add more tests in scenario test beside https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/computefleet/Azure.ResourceManager.ComputeFleet/tests/Scenario/ComputeFleetCRUDTests.cs

I have added more tests.

The assets push -test-proxy push -a .\sdk\computefleet\Azure.ResourceManager.ComputeFleet\assets.json

**Giving Error: - Please suggest**
git remote set-url origin https://github.com/Azure/azure-sdk-assets
StdOut:
StdErr:
ExitCode: 0
Unhandled exception: Azure.Sdk.Tools.TestProxy.Common.Exceptions.HttpException: Invocation of "git add -A ." had a non-zero exit code -1.
StdOut:
StdErr: The following paths and/or pathspecs matched paths that exist
outside of your sparse-checkout definition, so will not be
updated in the index:
net/SessionRec.json
hint: If you intend to update such entries, try one of the following:
hint: * Use the --sparse option.
hint: * Disable or modify the sparsity rules.
hint: Disable this message with "git config advice.updateSparsePath false"

rahuls-microsoft commented:
> > Since this is going GA, we should have a full test coverage for all operations, such as https://github.com/Azure/azure-sdk-for-net/blob/fb0b80d4c8fc99e9e964862f2a374d2bc28096f7/sdk/computefleet/Azure.ResourceManager.ComputeFleet/src/Generated/ComputeFleetResource.cs#L373C56-L373C87 Add more tests in scenario test beside https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/computefleet/Azure.ResourceManager.ComputeFleet/tests/Scenario/ComputeFleetCRUDTests.cs
> 
> I have added more tests.
> 
> The assets push -test-proxy push -a .\sdk\computefleet\Azure.ResourceManager.ComputeFleet\assets.json
> 
> **Giving Error: - Please suggest** git remote set-url origin https://github.com/Azure/azure-sdk-assets StdOut: StdErr: ExitCode: 0 Unhandled exception: Azure.Sdk.Tools.TestProxy.Common.Exceptions.HttpException: Invocation of "git add -A ." had a non-zero exit code -1. StdOut: StdErr: The following paths and/or pathspecs matched paths that exist outside of your sparse-checkout definition, so will not be updated in the index: net/SessionRec.json hint: If you intend to update such entries, try one of the following: hint: * Use the --sparse option. hint: * Disable or modify the sparsity rules. hint: Disable this message with "git config advice.updateSparsePath false"

@live1206 - Please suggest. Thanks for your help.

rahuls-microsoft committed:
Disable RecordedTest

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Merge branch 'Azure:main' into main

rahuls-microsoft committed:
Updated Test record upload

live1206 committed:
update assets

live1206 committed:
fix format

live1206 committed:
Update cspell.json

rahuls-microsoft committed:
Merge branch 'main' into main

live1206 committed:
Merge branch 'main' into rahuls-microsoft/main

live1206 committed:
update release date

rahuls-microsoft commented:
Can we release it next Tuesday - 10/22/2024?
The ARM Manifest rollout is in progress as a hotfix, so we should complete it in the next 2 days.

Thanks.

live1206 committed:
Merge branch 'main' of https://github.com/rahuls-microsoft/azure-sdk-for-net into rahuls-microsoft/main

SahilArora92 commented:
Please fix the build errors.

melina5656 committed:
Change type and name of some properties

melina5656 committed:
Update tsp-location.yaml

melina5656 committed:
Update tsp-location.yaml

live1206 commented:
/check-enforcer override
